test/lightning/accounts_test.exs

```diff
@@ -282,36 +284,120 @@ defmodule Lightning.AccountsTest do
         project_users: [%{user_id: user_2.id}]
       })
 
-      assert 2 == Repo.all(Credential) |> Enum.count()
       assert 2 == Repo.all(ProjectUser) |> Enum.count()
       assert 2 == Repo.all(User) |> Enum.count()
 
       :ok = Accounts.purge_user(user_1.id)
 
-      assert 1 == Repo.all(Credential) |> Enum.count()
       assert 1 == Repo.all(ProjectUser) |> Enum.count()
       assert 1 == Repo.all(User) |> Enum.count()
 
-      remaining_creds = Repo.all(Credential)
       remaining_projs = Repo.all(ProjectUser)
       remaining_users = Repo.all(User)
 
-      assert 1 == Enum.count(remaining_creds)
       assert 1 == Enum.count(remaining_projs)
       assert 1 == Enum.count(remaining_users)
 
-      assert remaining_creds
-             |> Enum.any?(fn x -> x.user_id == user_2.id end)
-
       assert remaining_projs
              |> Enum.any?(fn x -> x.user_id == user_2.id end)
 
+      refute remaining_projs
+             |> Enum.any?(fn x -> x.user_id == user_1.id end)
+
       assert remaining_users
              |> Enum.any?(fn x -> x.id == user_2.id end)
+
+      refute remaining_users
+             |> Enum.any?(fn x -> x.id == user_1.id end)
+    end
+
+    test "purging user sets all project credentials that use this user's credentials with nil" do
+      user = user_fixture()
+      project = Lightning.ProjectsFixtures.project_fixture()
+
+      project_credential_1 =
+        CredentialsFixtures.project_credential_fixture(
+          project_id: project.id,
+          user_id: user.id
+        )
+
+      project_credential_2 = CredentialsFixtures.project_credential_fixture()
+
+      job_1 =
+        JobsFixtures.job_fixture(project_credential_id: project_credential_1.id)
+
+      job_2 =
+        JobsFixtures.job_fixture(project_credential_id: project_credential_2.id)
+
+      refute job_1.project_credential_id |> is_nil()
+      refute job_2.project_credential_id |> is_nil()
+
+      :ok = Accounts.purge_user(user.id)
+
+      # job_1 project_credential_id is now set to nil
+      assert Jobs.get_job!(job_1.id).project_credential_id |> is_nil()
+
+      # while job_2 project_credential_id remain with a value
+      refute Jobs.get_job!(job_2.id).project_credential_id |> is_nil()
+    end
+
+    test "purging user deletes all project credentials that involve this user's credentials" do
+      user = user_fixture()
+
+      CredentialsFixtures.project_credential_fixture(user_id: user.id)
+      CredentialsFixtures.project_credential_fixture(user_id: user.id)
+      CredentialsFixtures.project_credential_fixture()
+
+      assert Credentials.count_project_credentials_for_user(user) == 2
```

Comment:
Regarding the function that was added, I think this can be written as:

```
from(pc in Ecto.assoc(user, [:credentials, :project_credentials]))
|> Repo.aggregate(:count, :id)
```

Reply:
You can put that in a function inside the test module for reuse.

Reply:
Done. Thanks